lib/application_json.rb

```diff
@@ -300,38 +300,24 @@ def reference_attributes
     ]
   end
 
-  def offer_json
-    JSON.pretty_generate(json_data['offer'])
+  def offer_conditions_json
+    JSON.pretty_generate(conditions: json_data['offer_conditions'])
   end
 
-  def offer_creation_json
-    course_details = json_data['offer']['course'].dup # we are going to mutate this :/
-    course_details.delete('description')
-    JSON.pretty_generate(
-      json_data['offer'].merge('course' => course_details)
-    )
-  end
-
-  def offer_attributes
+  def offer_conditions_attributes
     [
-      {
-        name: 'course',
-        type: link_to_resource_definition('Course'),
-        description: 'Course details'
-      },
-      {
-        name: 'date',
-        type: 'string',
-        description: 'The offered entry date in YYYY-MM-DD format'
-      },
       {
         name: 'conditions',
         type: 'array of strings [20]',
-        description: 'The conditions of the offer, limited to 20 conditions'
+        description: 'The conditions of the offer, limited to 20 conditions (optional)'
       }
     ]
   end
 
+  def unconditional_offer_json
```

Comment:
:+1: nice to keep JSON generation wrapped up in here, even when it's trivial as in this case

Reply:
Cool, thanks for the feedback! 😄 I was unsure whether or not this was silly. It made sense to me but didn't know what others might think.